aten/src/ATen/DLConvertor.cpp

```diff
@@ -90,13 +90,17 @@ DLDevice getDLDevice(const Tensor& tensor, const int64_t& device_id) {
     case DeviceType::HIP:
       ctx.device_type = DLDeviceType::kDLROCM;
       break;
+    case DeviceType::XPU:
+      ctx = at::detail::getXPUHooks().getDLPackDeviceFromATenDevice(
```

Comment:
Why is the indirection needed here? All the other types treat the device id uniformly across all types.

Reply:
This is because of the SYCL supporting the hierarchized device partitioning.
The DLPack device id is not same as the torch device id and vise versa.
All the relationship of the partitioning and address is encoded in the DLPack device id and address.

For the kDLoneAPI case, we need to map the different partitioning of the devices used by different framework by some specific mechanism in the hooks. That is not common for all other DLPack device type.